File: targets/java-spring/controllers/FileController.java
package com.baddemo.controllers;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import com.baddemo.services.FileService;

@RestController
@RequestMapping("/files")
public class FileController {

    private final FileService fileService = new FileService();

    @GetMapping("/download")
    public String download(@RequestParam String name) throws Exception {
        return fileService.readUserFile(name);
    }

    @GetMapping("/download_safe")
    public String downloadSafe(@RequestParam String name) throws Exception {
        return fileService.readWhitelisted(name);
    }
}

File: targets/php-laravel/app/Http/Controllers/FileController.php
<?php

namespace App\Http\Controllers;

use App\Services\FileService;
use Illuminate\Http\Request;

class FileController
{
    private FileService $files;

    public function __construct()
    {
        $this->files = new FileService();
    }

    public function download(Request $request)
    {
        $name = $request->query('name', '');
        $content = $this->files->readUserFile($name);
        return response()->json(['content' => $content]);
    }

    public function downloadV2(Request $request)
    {
        $name = $request->query('name', '');
        try {
            $content = $this->files->readWhitelisted($name);
        } catch (\InvalidArgumentException $e) {
            return response()->json(['error' => 'file not allowed'], 400);
        }
        return response()->json(['content' => $content]);
    }
}
